src/server/req_message.c

```diff
@@ -204,43 +204,14 @@ req_py_spawn(struct batch_request *preq)
 		return;
 	}
 
-	if (jt == IS_ARRAY_NO) {		/* a regular job is okay */
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {		/* a regular job or a subjob is okay */
 		/* the job must be running */
 		if ((pjob->ji_qs.ji_state != JOB_STATE_RUNNING) ||
 			(pjob->ji_qs.ji_substate !=
 			JOB_SUBSTATE_RUNNING)) {
 			req_reject(PBSE_BADSTATE, 0, preq);
```

Comment:
I think PBSE_UNKJOBID also need to be handled here.

Reply:
It is handled inside chk_job_request()